File: Backend/TopSys.TopConWeb.Domain/Entities/Tarefa.cs
using System;


namespace TopSys.TopConWeb.Domain.Entities
{
    public class Tarefa
    {
        public int Codigo { get; set; }
        public string Usuario { get; set; }
        public string Descricao { get; set; }
        public bool DiaInteiro { get; set; }
        public DateTime? Data { get; set; }
        public TimeSpan? Horario { get; set; }
        public string Contato { get; set; }
        public int DddTelefone { get; set; }
        public int Telefone { get; set; }
        public int DddCelular { get; set; }
        public int Celular { get; set; }
        public string Email { get; set; }
        public bool Finalizado { get; set; }
        public string Observacao { get; set; }
        public string Providencia { get; set; }
        public string Conclusao { get; set; }
        public int UsinaCodigo { get; set; }
        public int AnoVisita { get; set; }
        public int NumeroVisita { get; set; }
        public int AnoLead { get; set; }
        public int NumeroLead { get; set; }
        public int AnoOportunidade { get; set; }
        public int NumeroOportunidade { get; set; }
        public DateTime DataCriacao { get; set; }
        public string IdAgrupamento { get; set; }
    }
}

File: Frontend/src/app/classes/agenda/tarefa.ts
export class Tarefa {
    codigo: number = 0;
    usuario: string = '';
    descricao: string = '';
    diaInteiro: boolean;
    data: Date = new Date();

    horario: string = '';
    horarioString: string = '';

    observacao: string = '';
    contato: string = '';
    dddTelefone: number = 0; 
    telefone: number = 0;
    dddCelular: number = 0; 
    celular: number = 0;
    email: string = '';
    finalizado: boolean;
    providencia: string = '';
    conclusao: string = '';
    usinaCodigo: number = 0;
    anoVisita: number = 0;
    numeroVisita: number = 0;
    anoLead: number = 0;
    numeroLead: number = 0;
    anoOportunidade: number = 0;
    numeroOportunidade: number = 0;
    dataCriacao: Date = new Date();
    idAgrupamento: string = '';

    public clone(): Tarefa {
        const novaTarefa = new Tarefa();
        Object.assign(novaTarefa, this);
        return novaTarefa;
    }

}